### DISCUSSION TITLE
Is it always safe to use 'ref' in Notifier?

### DISCUSSION THREAD

**Discussion by fujidaiti** [rrousselGit/riverpod#2861] [Q&A]
Hello, I appreciate your great works :)

My question is, is it always safe to use `ref.read` in methods other than `build` of [Notifier](https://pub.dev/documentation/riverpod/latest/riverpod/Notifier-class.html) class like:

```dart
class Counter extends Notifier<int> {
  @override
  int build() {
    return 0;
  }

  void doSomething() {
    final value = ref.read(anotherProvider);
    // Update the state with 'value'
  }
}
```

Or is there a way to check if the `ref` is available or not like the `mounted` flag of StatefulWidget?

**Comment by Naveed-dev30** [comment]
Counting on this question.
